### PR TITLE
IGNITE-9284 Add standard scaler

### PR DESCRIPTION
**Pull request by dehasi**

### CODE REVIEW

**Comment by dehasi** [comment]
@zaleslaw @ybabak could you take a look please

File: modules/ml/src/main/java/org/apache/ignite/ml/preprocessing/standardscaling/StandardScalerTrainer.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                    return a.merge(b);
+                });
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /** A Service class which used for sums holing. */
+    private static class SumHelper implements AutoCloseable {
```
Comment (avplatonov) [inline modules/ml/src/main/java/org/apache/ignite/ml/preprocessing/standardscaling/StandardScalerTrainer.java]: Can you separate SumHelper and StandardScalerTrainer (especially in context that close() method is empty)?
It seems that SumHelper class may be usefull in other code.

File: modules/ml/src/main/java/org/apache/ignite/ml/preprocessing/standardscaling/StandardScalerPreprocessor.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+ *
+ * where {@code i} is a number of column, {@code mean_i} is the mean value this column and {@code sigma_i} is the
+ * standard deviation in this column.
+ *
+ * @param <K> Type of a key in {@code upstream} data.
+ * @param <V> Type of a value in {@code upstream} data.
+ */
+public class StandardScalerPreprocessor<K, V> implements IgniteBiFunction<K, V, Vector> {
+    /** */
+    private static final long serialVersionUID = 1L;
```
[the remark this answers is not in the document]
Comment (dehasi) [reply modules/ml/src/main/java/org/apache/ignite/ml/preprocessing/standardscaling/StandardScalerPreprocessor.java]: I generated UID.

File: modules/ml/src/main/java/org/apache/ignite/ml/preprocessing/standardscaling/StandardScalerTrainer.java (hunk 1 above)
Comment (dehasi) [reply modules/ml/src/main/java/org/apache/ignite/ml/preprocessing/standardscaling/StandardScalerTrainer.java]: I extracted `SumHelper` to `StandardScalerData`.

**Comment by dehasi** [comment]
@avplatonov thank you for the review. I made changes.

**Pull request closed without merging**